ios/LayoutReanimation/REASharedTransitionManager.m

```diff
@@ -0,0 +1,373 @@
+    }
+    if ([view1.reactTag intValue] < [view2.reactTag intValue]) {
+      return (NSComparisonResult)NSOrderedDescending;
+    }
+    return (NSComparisonResult)NSOrderedSame;
+  }];
+}
+
+- (NSMutableArray<REASharedElement *> *)getSharedElementForCurrentTransition:(NSArray *)sharedViews
+                                                             withNewElements:(BOOL)withNewElements
+{
+  NSMutableSet<NSNumber *> *viewTags = [NSMutableSet new];
+  if (!withNewElements) {
+    for (UIView *view in sharedViews) {
+      [viewTags addObject:view.reactTag];
+    }
+  }
+  NSMutableArray<REASharedElement *> *sharedElements = [NSMutableArray new];
+  for (UIView *sharedView in sharedViews) {
+    NSNumber *targetViewTag = _findTheOtherForSharedTransition(sharedView.reactTag);
+    bool bothAreRemoved = !withNewElements && [viewTags containsObject:targetViewTag];
+    if (targetViewTag == nil) {
+      // the sibling of shared view doesn't exist yet
+      continue;
+    }
+    UIView *viewSource;
+    UIView *viewTarget;
+    if (withNewElements) {
+      viewSource = [_animationManager viewForTag:targetViewTag];
+      viewTarget = sharedView;
+      UIView *viewTargetParentScreen = [self getParentScreen:viewTarget];
+      if (viewTargetParentScreen != nil) {
+        [self observeChanges:viewTargetParentScreen];
+      }
+      UIView *viewSourceParentScreen = [self getParentScreen:viewSource];
+      if (viewSourceParentScreen != nil) {
+        [self unobserveChanges:viewSourceParentScreen];
+      }
+    } else {
+      viewSource = sharedView;
+      viewTarget = [_animationManager viewForTag:targetViewTag];
+    }
+    if (bothAreRemoved) {
+      // case for nested stack
+      [self clearAllSharedConfigsForView:viewSource];
+      [self clearAllSharedConfigsForView:viewTarget];
+      continue;
+    }
+
+    REASnapshot *sourceViewSnapshot = [[REASnapshot alloc] init:viewSource withParent:viewSource.superview];
+    REASnapshot *targetViewSnapshot = [[REASnapshot alloc] init:viewTarget withParent:viewTarget.superview];
+    _snapshotRegistry[viewSource.reactTag] = sourceViewSnapshot;
+    _snapshotRegistry[viewTarget.reactTag] = targetViewSnapshot;
+
+    [_currentSharedTransitionViews addObject:viewSource];
+    [_currentSharedTransitionViews addObject:viewTarget];
+
+    REASharedElement *sharedElement = [[REASharedElement alloc] initWithSourceView:viewSource
+                                                                sourceViewSnapshot:sourceViewSnapshot
+                                                                        targetView:viewTarget
+                                                                targetViewSnapshot:targetViewSnapshot];
+    [sharedElements addObject:sharedElement];
+  }
+  return sharedElements;
+}
+
+- (UIView *)getParentScreen:(UIView *)view
+{
+  UIView *screen = view;
+  while (![NSStringFromClass([screen class]) isEqualToString:@"RNSScreenView"] && screen.superview != nil) {
```

Comment:
Is this really the best way of checking if the view is `RNSScreenView`?
Why not use `isMemberOfClass:` or `isKindOfClass:`?